test/unit-tests/adapters/axelar/libraries/StringAddressConversion.t.sol

```diff
@@ -0,0 +1,117 @@
+        );
+    }
+
+    /// @dev tests conversion of string to address
+    function testToAddress() public {
+        string memory testString = "0x1234567890123456789012345678901234567890";
+        address result = conversionHelper.toAddress(testString);
+        address expected = address(0x1234567890123456789012345678901234567890);
+
+        assertTrue(result == expected, "Converted address does not match expected value");
+    }
+
+    /// @dev tests invalid address conversion
+    function testInvalidAddressStringConversion() public {
+        string memory invalidAddressString = "1234567890123456789012345678901234567892";
+
+        bytes4 selector = bytes4(keccak256(bytes("InvalidAddressString()")));
+        vm.expectRevert(selector);
+        conversionHelper.toAddress(invalidAddressString);
+    }
+
+    /// @dev tests short address string
+    function testShortAddressStringConversion() public {
+        string memory shortAddressString = "0x12345678901234567890123456789012345678";
+
+        bytes4 selector = bytes4(keccak256(bytes("InvalidAddressString()")));
+        vm.expectRevert(selector);
+        conversionHelper.toAddress(shortAddressString);
+    }
+
+    /// @dev tests long address string
+    function testLongAddressStringConversion() public {
+        string memory longAddressString = "0x123456789012345678901234567890123456789012";
+
+        bytes4 selector = bytes4(keccak256(bytes("InvalidAddressString()")));
+        vm.expectRevert(selector);
+        conversionHelper.toAddress(longAddressString);
+    }
+
+    /// @dev tests invalid prefix in address string
+    function testInvalidPrefixAddressStringConversion() public {
+        string memory invalidPrefixAddressString = "1x1234567890123456789012345678901234567890";
+
+        bytes4 selector = bytes4(keccak256(bytes("InvalidAddressString()")));
+        vm.expectRevert(selector);
+        conversionHelper.toAddress(invalidPrefixAddressString);
+    }
+
+    /// @dev tests address string with invalid characters
+    function testInvalidCharacterAddressStringConversion() public {
+        string memory invalidCharacterAddressString = "0x12345678901234567890123456789012345678g0"; // 'g' is an invalid character
+
+        bytes4 selector = bytes4(keccak256(bytes("InvalidAddressString()")));
+        vm.expectRevert(selector);
+        conversionHelper.toAddress(invalidCharacterAddressString);
+    }
+
+    /// @dev tests conversion of string with lowercase hex characters to address
+    function testLowercaseHexCharacterToAddress() public {
+        string memory testString = "0xabcdefabcdefabcdefabcdefabcdefabcdefabcd";
+        address result = conversionHelper.toAddress(testString);
+        address expected = address(0xABcdEFABcdEFabcdEfAbCdefabcdeFABcDEFabCD);
+
+        assertTrue(result == expected, "Converted address with lowercase hex characters does not match expected value");
```

Comment:
nit: can we use `assertEq` in place of `assertTrue` here and in similar instances, as noted earlier